app/jobs/result_monitor_loader.rb

```diff
@@ -0,0 +1,22 @@
+# Load a result from S3 into the db
+require 'logger'
+class ResultMonitorLoader
+  @queue = :q03_pipeline_run
+  @logger = Logger.new(STDOUT)
+
+  def self.perform(pipeline_run_id, output)
+    @logger.info("Loading #{output} for pipeline run #{pipeline_run_id}")
+    pr = PipelineRun.find(pipeline_run_id)
+    return unless pr.result_status_for(output) == PipelineRun::STATUS_LOADING_QUEUED
+    begin
+      pr.update_result_status(output, PipelineRun::STATUS_LOADING)
+      pr.send(PipelineRun::LOADERS_BY_OUTPUT[output])
+      pr.update_result_status(output, PipelineRun::STATUS_LOADED)
+    rescue
+      pr.update_result_status(output, PipelineRun::STATUS_FAILED)
+      pr.update(results_finalized: 1)
```

Comment:
What happens if postprocessing result loading fails before alignmnet result loading has started running on a resque worker?

Reply:
The result monitor stops polling for updates and the alignment result loading will complete eventually